### ISSUE TITLE
[BUG/ISSUE] GCHP setRestartLink.sh run directory script assumes incorrect restart filename when restarting simulation

### ISSUE THREAD

**Issue by msulprizio** [geoschem/geos-chem#1866]
### Name and Institution (Required)

Name: Melissa Sulprizio
Institution: Harvard / GCST

### Confirm you have reviewed the following documentation

- [x] [Support guidelines](https://geos-chem.readthedocs.io/en/stable/reference/SUPPORT.html)
- [x] [User manuals](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/related-docs.html)
- [x] [Debugging GEOS-Chem and HEMCO errors](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/debug-guide.html)
- [x] [Current and past Github issues](https://github.com/geoschem/geos-chem/issues)

### Description of your issue or question

I am using the current 14.2.0 development branch (tag 14.2.0-rc.0). When resuming a GCHP simulation from a previous run that crashed, I received the following error:

```
ERROR: Unable to set symlink gchp_restart.nc4 because file ./Restarts/GEOSChem.Restart.20190501_0000z.c24.nc4 does not exist! Create file or link with that name, or change start date in cap_restart and/or grid resolution in setCommonRunSettings.sh to match restart file that exists.
``` 

The issue is that script `setRestartLink.sh` assumes the restart file is in the format `rst_target=./Restarts/GEOSChem.Restart.${start_str:0:13}z.c${N}.nc4`. However, restart files saved out from GCHP simulations do not contain the cube-sphere resolution string (`c${N}`) in the filename. For example, the default name of files saved out are:

```
[Restarts]$ ls
gcchem_internal_checkpoint.20190201_0000z.nc4
gcchem_internal_checkpoint.20190301_0000z.nc4
gcchem_internal_checkpoint.20190401_0000z.nc4
gcchem_internal_checkpoint.20190501_0000z.nc4
GEOSChem.Restart.20190101_0000z.c24.nc4
GEOSChem.Restart.20190101_0000z.nc4
```

One solution would be to remove the cubesphere resolution from the filename in `setRestartLink.sh` but this would have implications for the out-of-the-box GCHP simulations, which include symbolic links to restart files for several cube-sphere resolutions to allow for flexibility in choosing the simulation resolution.

**Comment by msulprizio** [comment]
@lizziel Any thoughts on a preferred fix?

**Comment by msulprizio** [comment]
It looks like we do have code to rename the checkpoint file to the correct file name in the sample run scripts:

https://github.com/geoschem/geos-chem/blob/4722f288e90291ba904222f4bbe4fc216d17c34a/run/GCHP/runScriptSamples/gchp.batch_job.sh#L154-L156

This line only gets executed if the simulation finished successfully though. Perhaps this is fine, and we simply require users to manually create the restart file symbolic link. 

We could also modify `setRestartLink.sh` to accept an optional argument for the restart/checkpoint file to use (e.g. `setRestartLink.sh Restarts/gcchem_internal_checkpoint.20190501_0000z.nc4`) and that would point the `gchp_restart.nc4` symbolic link to the specified file rather than the default one.

**Comment by msulprizio** [comment]
Note: I also needed to comment out the execution of `setRestartLinks.sh` in my run script to avoid overwriting the symbolic link that I manually created.

**Comment by lizziel** [comment]
It looks like the issue is renaming the mid-run checkpoints for the case of the run crashing. Is that right? If yes, we can put the renaming of the mid-run checkpoints before the end-of-run restart renaming and restart symbolic link update. So that would put [this](https://github.com/geoschem/geos-chem/blob/dev/14.2.0/run/GCHP/runScriptSamples/gchp.batch_job.sh#L159-L173) code before [this](https://github.com/geoschem/geos-chem/blob/0f3097d2c99852b1668c2be3e7786aaa19f11a9a/run/GCHP/runScriptSamples/gchp.batch_job.sh#L147-L157).

If you then want to start the run at a new time (such as if the run just crashed, so start at a previous month) then you should be able to change only the date in [cap_restart](url) and not have to manually set the restart symbolic link.

**Comment by msulprizio** [comment]
@lizziel Yes, I think that solution should work.

**Comment by msulprizio** [comment]
The fix for this (PR #1868) has now been merged into the 14.2.0 development branch.
